js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx

```diff
@@ -16,25 +14,27 @@ export const HelpLinkTextInput = ({
   url,
 }: HelpLinkTextInputProps) => {
   const { t } = useTranslation();
-  const { register } = useFormContext();
-  const name = fieldName.substr(fieldName.indexOf(".") + 1);
+  const form = useFormContext();
+  const name = fieldName.slice(fieldName.indexOf(".") + 1);
   return (
-    <FormGroup
-      label={t(name)}
-      fieldId={name}
-      labelIcon={
-        <HelpItem
-          helpText={
-            <Trans i18nKey={`${name}Help`}>
-              Default value prevents pages from being included
-              <FormattedLink href={url} title={t("learnMore")} />
-            </Trans>
-          }
-          fieldLabelId={name}
-        />
-      }
-    >
-      <KeycloakTextInput id={name} {...register(fieldName)} />
-    </FormGroup>
+    <FormProvider {...form}>
+      <FormGroup
+        label={t(name)}
+        fieldId={name}
+        labelIcon={
+          <HelpItem
+            helpText={
+              <Trans i18nKey={`${name}Help`}>
+                Default value prevents pages from being included
+                <FormattedLink href={url} title={t("learnMore")} />
+              </Trans>
+            }
+            fieldLabelId={name}
+          />
+        }
+      >
+        <TextControl name={fieldName} />
```

Comment:
A `TextControl` already includes a `FormGroup`, so I don't think this is the right use-case. Either we replace the while `FormGroup` above, or we leave this unported and handle it manually in the PF5 upgrade.

Reply:
Thanks @jonkoops maybe let's leave it and handle it manually in the PF5 upgrade.

Reply:
Sure!